mythril_core/src/device/mod.rs

```diff
@@ -340,4 +347,75 @@ mod test {
         assert_eq!([0x12, 0x34], val.as_slice());
         assert_eq!(0x1234, val.as_u32());
     }
+
+    #[test]
+    fn test_conflicting_portio_device() {
+        let mut map = DeviceMap::default();
+        let com = ComDevice::new(0, 0);
+        map.register_device(com).unwrap();
+        let com = ComDevice::new(0, 0);
+
+        assert!(map.register_device(com).is_err());
+    }
+
+    #[test]
+    fn test_overlapping_portio_devices() {
+        // This is just a dummy device so we can have arbitrary port ranges
+        // for testing.
+        struct DummyDevice {
```

Comment:
I would move these out of the function, up in to the `mod test`, and then make the different `assert`s in to different tests